examples/error-type-intersection/error_type_intersection.md

```diff
@@ -0,0 +1,7 @@
+# Type intersection for error types
+
+You can define an error type that is both a subtype of a `distinct` error type and has additional constraints on the detail fields using intersection types.
```

Comment:
```suggestion
You can define an error type that is a subtype of a `distinct` error type and has additional constraints on the detail fields using intersection types.
```

Reply:
Any particular reason to remove "both"?

Reply:
One is a type, and the other is a constraint on the type. Does both match here? Fine to keep it too

Reply:
What we are trying to say is 

```
You can define an error type
- that is a subtype of a `distinct` error type
- that has additional constraints on the detail fields
```

So "both" here is OK IMO.